src/main/java/org/opensearch/ad/rest/handler/InsightsJobActionHandler.java

```diff
@@ -0,0 +1,476 @@
+            client.get(getRequest, ActionListener.wrap(response -> {
+                if (response.isExists()) {
+                    // Job exists, check if it's already enabled
+                    try (
+                        XContentParser parser = RestHandlerUtils
+                            .createXContentParserFromRegistry(xContentRegistry, response.getSourceAsBytesRef())
+                    ) {
+                        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.nextToken(), parser);
+                        Job existingJob = Job.parse(parser);
+
+                        if (existingJob.isEnabled()) {
+                            logger.info("Insights job is already running");
+                            listener.onResponse(new InsightsJobResponse("Insights job is already running"));
+                            return;
+                        }
+
+                        // Use current user if provided, otherwise keep existing user (for BWC)
+                        Job enabledJob = new Job(
+                            existingJob.getName(),
+                            createSchedule(frequency),
+                            existingJob.getWindowDelay(),
+                            true,
+                            Instant.now(),
+                            null,
+                            Instant.now(),
+                            existingJob.getLockDurationSeconds(),
+                            user != null ? user : existingJob.getUser(),
+                            existingJob.getCustomResultIndexOrAlias(),
+                            existingJob.getAnalysisType()
+                        );
+
+                        indexJob(
+                            enabledJob,
+                            listener,
+                            String.format(Locale.ROOT, "Insights job restarted successfully with frequency: %s", frequency)
+                        );
+
+                    } catch (IOException e) {
+                        logger.error("Failed to parse existing insights job", e);
+                        listener
+                            .onFailure(
+                                new OpenSearchStatusException("Failed to parse existing insights job", RestStatus.INTERNAL_SERVER_ERROR)
+                            );
+                    }
+                } else {
+                    createNewJob(frequency, user, listener);
+                }
+            }, e -> {
+                logger.error("Failed to check for existing insights job", e);
+                listener.onFailure(e);
+            }));
+        }
+    }
+
+    /**
+     * Create a brand new insights job.
+     */
+    private void createNewJob(String frequency, User user, ActionListener<InsightsJobResponse> listener) {
+        try {
+            IntervalSchedule schedule = createSchedule(frequency);
+            long lockDurationSeconds = java.time.Duration.of(schedule.getInterval(), schedule.getUnit()).getSeconds() * 2;
```

Comment:
lockDurationSeconds should be less than or equal to frequency, right?

Reply:
yes good catch, updated it to 
```
            long lockDurationSeconds = java.time.Duration.of(schedule.getInterval(), schedule.getUnit()).getSeconds();

```